packages/next/src/build/analysis/get-page-static-info.ts

```diff
@@ -229,6 +233,18 @@ function checkExports(
               )
           }
         }
+
+        if (node.type === 'ExpressionStatement') {
+          if (node.expression.type === 'StringLiteral') {
+            const directive = node.expression.value
+            if (CLIENT_DIRECTIVE_REGEX.test(directive)) {
```

Comment:
```suggestion
            if (directive === 'use client') {
```

Wouldn't it be faster to compare the string then the regex pattern?